fe/fe-core/src/main/java/org/apache/doris/qe/MasterOpExecutor.java

```diff
@@ -42,7 +42,9 @@
     // the total time of thrift connectTime add readTime and writeTime
     private int thriftTimeoutMs;
 
-    public MasterOpExecutor(OriginStatement originStmt, ConnectContext ctx, RedirectStatus status) {
+    private boolean shouldNotRetry;
+
+    public MasterOpExecutor(OriginStatement originStmt, ConnectContext ctx, RedirectStatus status, boolean isQuery) {
```

Comment:
isQuery rename to shouldRetry? and variable 'shouldNotRetry' in MasterOpExecutor rename to "shouldRetry"? I think it is more readable. and then we just need to think when need to retry

Reply:
I was discussed this name with my colleague, the constructor of MasterOpExecutor expose 'shouldRetry' could lead to misuse, because could not remember the retry related to statement type when invoke `new MasterOpExecutor()`.

Although `isQuery` doesn't look good, but it always write collect.

And `shouldNotRetry` means exactly that: we should not retry forward to master

Reply:
If we want retry a statement, just need it is a idempotent operation, not just query, right?

Reply:
yes，the condition isQuery is conservative & convenient

Reply:
the principle is：we must ensure idempotent operation execute at-most-once

Reply:
Thank You, I see